MoEngageSegmentPluginBase/MoEngageSegmentPluginBase/MoEngageSegmentPluginParser.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  MoEngageSegmentPluginParser.swift
+//  MoEngageSegmentPluginBase
+//
+//  Created by Rakshitha on 09/05/23.
+//
+
+import Foundation
+import MoEngageSDK
+
+class MoEngageSegmentPluginParser {
```

Comment:
Do we really need to keep the methods static? I would always be in memory.

Reply:
@umangmoe  done